Index: mplayer/loader/qtx/qtxsdk/components.h
===================================================================
--- mplayer.orig/loader/qtx/qtxsdk/components.h	2012-07-23 01:28:15.670622100 +0200
+++ mplayer/loader/qtx/qtxsdk/components.h	2012-07-23 01:33:31.883912000 +0200
@@ -2,6 +2,8 @@
 #define MPLAYER_COMPONENTS_H
 
 #include <inttypes.h>
+#include "mp_msg.h"
+#define printf(...) mp_msg(MSGT_CPLAYER, MSGL_V, __VA_ARGS__)
 
 // Basic types:
 
@@ -751,5 +753,5 @@
 
 
 }
-
+#undef printf
 #endif /* MPLAYER_COMPONENTS_H */
